torchdynamo/symbolic_convert.py

```diff
@@ -359,7 +359,19 @@ def call_function(self, fn, args, kwargs):
                 assert forward is not torch.nn.Module.forward
                 self.inline_user_function(fn.guards, forward, [fn] + args, kwargs)
         elif isinstance(fn, UserFunctionVariable):
-            self.inline_user_function(fn.guards, fn.fn, fn.self_args() + args, kwargs)
+            if fn.fn.__class__ is type:
+                options = VariableTracker.propagate(
+                    [
+                        fn,
+                    ]
+                    + list(args)
+                    + list(kwargs.values())
+                )
+                self.push(UserClassInstanceVariable(fn.fn(*args, **kwargs), **options))
```

Comment:
Here you are calling the user code (fn.fn) at compile time.  We should never do this.  This won't work because
1) fn.fn() will only get called once and the result will be hardcoded in the graph (rather than the correct behavior of calling fn.fn() every time the function is called).
2) Calling fn.fn() here could cause arbitrary things to happen, we have no idea it what it does.  So we could lose control of process execution and not capture lots of stuff.

It might be easier to start out by not supporting constructors, and instead just try to support user classes passed as arguments.  These will show up as `UnsupportVariable()`.

When we do try to support constructors, we should leave the constructors in the python code.  I have some pending changes that will make doing something like this easier.